src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
How about an option "I send messages from this assembly" declaration? By default we load all the types in each endpoints and log `warn` in the current code.

But if a given endpoint includes the list of assemblies it "knows", we ignore all routes for types from different assemblies. Yes, it adds a bit of effort in the multiple contract assembly case but it saves you `warn`s that are false positives.

e.g.
```
<endpoints>
  <endpoint name="endpointA">
    <contracts>
       <contract assembly="MyContracts"/>
    </contracts>
    <handles>
      <command type="Contracts.Commands.DemoCommand, Contracts" />
      <event type="Contracts.Events.DemoEvent, Contracts" />
    </handles>
  </endpoint>
  <endpoint name="endpointb">
```